OutboxSample/Model/Events/EventMetadataAttribute.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace OutboxSample.Model.Events;
+
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct, AllowMultiple = false, Inherited = true)]
+public class EventMetadataAttribute : Attribute
```

Comment:
Minor: you might want to seal this attribute, as any other properties will be ignored by the static helper method.